### DISCUSSION TITLE
How to detect client side?  process is not defined

### DISCUSSION THREAD

**Discussion by edwh** [nuxt/nuxt#17081] [Questions] [answered]
In Nuxt2 I used to use `process.browser` within JS code to test whether I was on the client side.  I'm running SSR (`target: 'static'`, `ssr: true`) and getting client-side errors about `process` being undefined.

Perhaps this has changed, and I can't find it in the docs - apologies if that is the case.

**Comment by danielrjames** [comment]
`process.client` or `!process.server`

**Comment by edwh** [comment]
That's what used to work, but 'process' is undefined in on the client when SSR is used.

**Comment by edwh** [comment, marked as answer]
This seems to work in the latest release.

**Comment by ErickLuis00** [comment]
Alternative to process.<server, client, browser...>.

```javascript
const app = useNuxtApp();

// app.ssrContext is available only on the server side.
const isClient = !app.ssrContext

if (isClient) {
 // client side code
}

if (!isClient) {
 // server side code

 // You can set custom headers
 app.ssrContext.event.res.setHeader('X-My-Header', 'My Value');
}
 ```
It works with all nitro providers that don't support Nodejs, like Cloudflare Workers.
